### ISSUE TITLE
Construct 3 crashes when removing sprite from family and running debug.

### ISSUE THREAD

**Issue by scoobyabbott** [Scirra/Construct-3-bugs#669]
Construct 3 crashes when removing sprite from family and running debug.

When I remove any sprite from the EnemyFreeze family and run debug the program crashes.

https://drive.google.com/open?id=0B1IqTklMvM1reG9YZ0M2S3pSVzA

*A minimal Construct 3 project (.c3p) is required to be attached. Please see the guidelines*

## Steps to reproduce

1. Add sprite into family.
2. Remove sprite from family.
3. Run debug.

## Observed result
Construct 3 crashed.

*What do you see happen?*
Screen turned pink and the "oops something went wrong dialog popped up"

## Affected browsers

* Chrome: yes
* Firefox: n/a
* Edge: n/a
* Safari: n/a

## System details
OS Name	Microsoft Windows 10 Home
Version	10.0.15063 Build 15063
Other OS Description 	Not Available
OS Manufacturer	Microsoft Corporation
System Name	DESKTOP-0LA89SQ
System Manufacturer	Dell Inc.
System Model	XPS 8500
System Type	x64-based PC
System SKU	To be filled by O.E.M.
Processor	Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz, 3401 Mhz, 4 Core(s), 8 Logical Processor(s)
BIOS Version/Date	Dell Inc. A06, 5/25/2012
SMBIOS Version	2.7
Embedded Controller Version	255.255
BIOS Mode	Legacy
BaseBoard Manufacturer	Dell Inc.
BaseBoard Model	Not Available
BaseBoard Name	Base Board
Platform Role	Desktop
Secure Boot State	Unsupported
PCR7 Configuration	Binding Not Possible
Windows Directory	C:\WINDOWS
System Directory	C:\WINDOWS\system32
Boot Device	\Device\HarddiskVolume4
Locale	United States
Hardware Abstraction Layer	Version = "10.0.15063.0"
User Name	DESKTOP-0LA89SQ\shawn
Time Zone	Eastern Daylight Time
Installed Physical Memory (RAM)	16.0 GB
Total Physical Memory	16.0 GB
Available Physical Memory	7.88 GB
Total Virtual Memory	18.3 GB
Available Virtual Memory	7.37 GB
Page File Space	2.38 GB
Page File	C:\pagefile.sys
Device Encryption Support	Reasons for failed automatic device encryption: TPM is not usable, PCR7 binding is not supported, Hardware Security Test Interface failed and device is not InstantGo, Un-allowed DMA capable bus/device(s) detected, TPM is not usable
Hyper-V - VM Monitor Mode Extensions	Yes
Hyper-V - Second Level Address Translation Extensions	Yes
Hyper-V - Virtualization Enabled in Firmware	Yes
Hyper-V - Data Execution Protection	Yes

*If you see a crash report dialog, please copy and paste it to where it says "PASTE HERE" below. Otherwise please go to **Menu** -> **About** -> **Platform information** and paste that information there instead.*

<details><summary>View details</summary>

Error report information
Type: assertion failure
Message: cannot find instance variable export index
Stack: Error at Function.ǃp.ǃSEC (https://editor.construct.net/r47/main.js:2:202189) at t (https://editor.construct.net/r47/main.js:2:171852) at window.assert (https://editor.construct.net/r47/main.js:2:172255) at a.ǃpxb (https://editor.construct.net/r47/main.js:61:127968) at window.ǃAV.ǃpPv (https://editor.construct.net/r47/main.js:61:324965) at window.ǃAV.ǃpVc (https://editor.construct.net/r47/main.js:61:318987) at o.map (https://editor.construct.net/r47/main.js:61:285259) at Array.map () at e.ǃpVc (https://editor.construct.net/r47/main.js:61:285248) at t.map (https://editor.construct.net/r47/main.js:61:278778)
Construct 3 version: r47
URL: https://editor.construct.net/
Date: Fri Aug 04 2017 11:34:49 GMT-0400 (Eastern Daylight Time)
Uptime: 84.8 s

Platform information
Browser: Chrome
Browser version: 60.0.3112.90
Browser engine: Blink
Browser architecture: 64-bit
Context: browser
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1.100000023841858
Logical CPU cores: 8
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36
C3 release: r47

WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Vendor: Google Inc.
Renderer: ANGLE (AMD Radeon HD 7500 Series Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions: EXT_color_buffer_float, EXT_disjoint_timer_query_webgl2, EXT_texture_filter_anisotropic, OES_texture_float_linear, WEBGL_compressed_texture_s3tc, WEBGL_compressed_texture_s3tc_srgb, WEBGL_debug_renderer_info, WEBGL_debug_shaders, WEBGL_lose_context

</details>

**Comment by scoobyabbott** [comment]
Update: Okay, so it seems that when I removed the sprite from the family it kept the family instance variable in event sheet. So, when it ran it looked for an instance variable that wasn't there. 

When being removed from a family, should it also remove any instance variables from the event sheet?

**Comment by NetOneGames** [comment]
if the object is removed from the family it definitely should NOT keep the family instance variables,
but the family instance variables are not stored in the event sheet as per your comment
they are stored in the family object
so your comment dosent make 100% sense.  you need to confrim

**Comment by scoobyabbott** [comment]
Okay. So here's what I mean...I put the sprite in a family called "Frozen". Inside the family there is an instance variable called Frozen that is either true or false. In the event sheet I called for certain actions to take place if the instance variable is true. 

So what I thought happened when you removed a sprite from a family was it eliminated the instance variable from the sprite, but also deleted the call to action in the event sheet since they sprite no longer has the variable.

Make sense?

**Comment by NetOneGames** [comment]
For some reason I can't open your file on my phone. I'll have a look on pc in a bit.  
But generally ...
you can't delete all objects from a family, there always has to be at least one object in a family.  If you delete or remove any other objects from the family it will have no effect on the family. All family instance variables and behaviours and events will remain intact as they are linked to the family.  And  the sprite that you removed will no linger be affected by any events on that family and will no longer inheret it's variables behaviours etc.

**Comment by NetOneGames** [comment]
I have played around a bit with your project, cool project by the way!
I added my own sprite to the EnemyFreeze family ran the game fine OK
I deleted this sprite from the EnemyFreeze family and then run debug mode It ran fine again OK
I also removed some of your own enemy sprites from the same family and there was no problems.
Adding them back in again,  no problems game ran OK.

When you get your error I think you are deleting the sprite directly from the EnemyFreeze Family tree instead of using the family editor dialogue ? 

if so then this is deleting it from the game entirely might be causing issues as it will delete all actions associated with it which could in turn crash other parts of the game looking for those actions as well.
I think this is what is happening.

use the family edit dialog to add and remove enemies from you EnemyFreeze family   (or other families)
